### DISCUSSION TITLE
503 service unavailable

### DISCUSSION THREAD

**Discussion by Shan2628** [tryzealot/zealot#1255] [Q&A] [answered]
<img width="591" alt="Screenshot 2023-08-24 at 17 19 48" src="https://github.com/tryzealot/zealot/assets/141384753/aacbd0e5-08eb-401a-a897-c565cc0a5149">
Is the app under maintenance?

**Comment by icyleaf** [comment]
VPS 小水管不抗造 🫣

**Reply by Shan2628** [reply]
对不起？你能帮我看看这里的问题是什么吗？

**Reply by Shan2628** [reply]
<img width="626" alt="Screenshot 2023-08-25 at 10 40 52" src="https://github.com/tryzealot/zealot/assets/141384753/9350f4ab-5e4a-41ad-a7d3-78ab7e7a3473">

**Reply by icyleaf** [reply]
这是只部署了 zealot 服务的内存情况吗？开局一张图啥也解决不了呢，起码按照[反馈模板](https://github.com/tryzealot/zealot/issues/new?assignees=&labels=bug&projects=&template=bug_report.yaml&title=%5BBug%5D%3A+) 把信息告知下？

**Reply by Shan2628** [reply]
WARNING: table: "settings" does not exist or not database connection, `Setting.keep_uploads` fallback to returns the default value.
2023-08-24T10:38:39.996Z pid=7981 tid=1px INFO: Sidekiq 7.1.2 connecting to Redis with options {:size=>10, :pool_name=>"internal", :url=>"redis://localhost:6379/0", :username=>nil, :password=>nil, :network_timeout=>5}
Address not available - connect(2) for [::1]:6379
/usr/local/lib/ruby/3.0.0/socket.rb:1214:in `__connect_nonblock'
/usr/local/lib/ruby/3.0.0/socket.rb:1214:in `connect_nonblock'
/usr/local/lib/ruby/3.0.0/socket.rb:56:in `connect_internal'
/usr/local/lib/ruby/3.0.0/socket.rb:137:in `connect'
/usr/local/lib/ruby/3.0.0/socket.rb:642:in `block in tcp'
/usr/local/lib/ruby/3.0.0/socket.rb:227:in `each'
/usr/local/lib/ruby/3.0.0/socket.rb:227:in `foreach'
/usr/local/lib/ruby/3.0.0/socket.rb:632:in `tcp'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client/ruby_connection.rb:124:in `connect'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client/ruby_connection.rb:49:in `initialize'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client.rb:685:in `new'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client.rb:685:in `block in connect'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client/middlewares.rb:12:in `connect'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client.rb:684:in `connect'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client.rb:671:in `raw_connection'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client.rb:638:in `ensure_connected'
/app/vendor/bundle/ruby/3.0.0/gems/redis-client-0.15.0/lib/redis_client.rb:218:in `call'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/lib/sidekiq/redis_client_adapter.rb:30:in `method_missing'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-scheduler-5.0.3/lib/sidekiq-scheduler/redis_manager.rb:238:in `block in hdel'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/lib/sidekiq/config.rb:164:in `block in redis'
/app/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:110:in `block (2 levels) in with'
/app/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:109:in `handle_interrupt'
/app/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:109:in `block in with'
/app/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:106:in `handle_interrupt'
/app/vendor/bundle/ruby/3.0.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:106:in `with'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/lib/sidekiq/config.rb:161:in `redis'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/lib/sidekiq.rb:73:in `redis'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-scheduler-5.0.3/lib/sidekiq-scheduler/redis_manager.rb:238:in `hdel'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-scheduler-5.0.3/lib/sidekiq-scheduler/redis_manager.rb:78:in `remove_job_schedule'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-scheduler-5.0.3/lib/sidekiq-scheduler/schedule.rb:115:in `remove_schedule'
/app/config/initializers/sidekiq_scheduled.rb:5:in `add_or_delete_schedule'
/app/config/initializers/sidekiq_scheduled.rb:38:in `block in <main>'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:445:in `instance_exec'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:445:in `block in make_lambda'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:199:in `block (2 levels) in halting'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:687:in `block (2 levels) in default_terminator'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:686:in `catch'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:686:in `block in default_terminator'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:200:in `block in halting'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:595:in `block in invoke_before'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:595:in `each'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:595:in `invoke_before'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:106:in `run_callbacks'
/app/vendor/bundle/ruby/3.0.0/gems/activesupport-7.0.7/lib/active_support/reloader.rb:88:in `prepare!'
/app/vendor/bundle/ruby/3.0.0/gems/railties-7.0.7/lib/rails/application/finisher.rb:68:in `block in <module:Finisher>'
/app/vendor/bundle/ruby/3.0.0/gems/railties-7.0.7/lib/rails/initializable.rb:32:in `instance_exec'
/app/vendor/bundle/ruby/3.0.0/gems/railties-7.0.7/lib/rails/initializable.rb:32:in `run'
/app/vendor/bundle/ruby/3.0.0/gems/railties-7.0.7/lib/rails/initializable.rb:61:in `block in run_initializers'
/usr/local/lib/ruby/3.0.0/tsort.rb:228:in `block in tsort_each'
/usr/local/lib/ruby/3.0.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
/usr/local/lib/ruby/3.0.0/tsort.rb:431:in `each_strongly_connected_component_from'
/usr/local/lib/ruby/3.0.0/tsort.rb:349:in `block in each_strongly_connected_component'
/usr/local/lib/ruby/3.0.0/tsort.rb:347:in `each'
/usr/local/lib/ruby/3.0.0/tsort.rb:347:in `call'
/usr/local/lib/ruby/3.0.0/tsort.rb:347:in `each_strongly_connected_component'
/usr/local/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
/usr/local/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
/app/vendor/bundle/ruby/3.0.0/gems/railties-7.0.7/lib/rails/initializable.rb:60:in `run_initializers'
/app/vendor/bundle/ruby/3.0.0/gems/railties-7.0.7/lib/rails/application.rb:372:in `initialize!'
/app/config/environment.rb:7:in `<top (required)>'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/lib/sidekiq/cli.rb:303:in `require'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/lib/sidekiq/cli.rb:303:in `boot_application'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/lib/sidekiq/cli.rb:42:in `run'
/app/vendor/bundle/ruby/3.0.0/gems/sidekiq-7.1.2/bin/sidekiq:31:in `<top (required)>'
./bin/sidekiq:29:in `load'
./bin/sidekiq:29:in `<main>'
Starting Zealot Worker
WARNING: table: "settings" does not exist or not database connection, `Setting.keep_uploads` fallback to returns the default value.
2023-08-24T10:38:46.864Z pid=7987 tid=1pn INFO: Sidekiq 7.1.2 connecting to Redis with options {:size=>10, :pool_name=>"internal", :url=>"redis://localhost:6379/0", :username=>nil, :password=>nil, :network_timeout=>5}
Address not available - connect(2) for [::1]:6379
/usr/local/lib/ruby/3.0.0/socket
<img width="575" alt="Screenshot 2023-08-25 at 11 07 05" src="https://github.com/tryzealot/zealot/assets/141384753/eeb7c511-bf03-444e-a20c-a090bd70695f">
.rb:1214:in `__connect_nonblock'
/usr/local/lib/ruby/3.0.0/socket.rb:1214:in `connect_nonblock'
/usr/local/lib/ruby/3.0.0/socket.rb:56:in `connect_internal'



This is my error..

**Reply by Shan2628** [reply]
xiexie

**Reply by icyleaf** [reply]
```diff
2023-08-24T10:38:39.996Z pid=7981 tid=1px INFO: Sidekiq 7.1.2 connecting to Redis with options {:size=>10, :pool_name=>"internal", :url=>"redis://localhost:6379/0", :username=>nil, :password=>nil, :network_timeout=>5}
+ Address not available - connect(2) for [::1]:6379
/usr/local/lib/ruby/3.0.0/socket.rb:1214:in __connect_nonblock' /usr/local/lib/ruby/3.0.0/socket.rb:1214:in connect_nonblock'
```

我看你设置了一个自己的 redis conf 你确认下里面的端口号是否是 6379

**Reply by icyleaf** [reply, marked as answer]
我创建了一个 telegram 讨论群，如果需要实时解决可进来聊 https://t.me/+csa3Y2KOx44wMGRl

**Reply by Shan2628** [reply]
thank you I will join

**Reply by Shan2628** [reply]
已加入等待接受
